engine/buff/sc_buff.cpp

```diff
@@ -2374,7 +2374,14 @@ void buff_t::aura_gain()
     {
       if ( !player->is_sleeping() )
       {
-        sim->print_log( "{} gains {} (value={})", *player, buff_display_name, current_value );
+        if ( sim->log_spell_id )
+        {
+          sim->print_log( "{} gains {} ({}) (value={})", *player, buff_display_name, s_data->id(), current_value );
+        }
+        else
+        {
+          sim->print_log( "{} gains {} (value={})", *player, buff_display_name, current_value );
+        }
```

Comment:
```suggestion
        std::string buff_display_name = fmt::format("{} (stacks={})", *this, current_stack );
        sim->print_log( "{} gains {} (value={})", *player, buff_display_name, current_value );
```

This requires changing line 2371 as well.

Should give a line like:
Player X gains Buff Y (spell_id) (current_stack=N) (value=x)